features/step_definitions/hooks-js.js

```diff
@@ -0,0 +1,42 @@
+const { spawn } = require('child_process');
+const { expect } = require('chai');
+const { Given, When, Then } = require('cucumber');
+
+
+Given('an API description with transactions "GET /articles" and "POST /articles"', function () {
+  this.dredd.apiDescription = `test/fixtures/blog/apidesc${this.apiDescriptionExt}`;
+});
+
+Given('an implementation, which requires authorization on "POST /articles"', function () {
+  this.dredd.args.push('--server=node test/fixtures/blog/app.js');
+});
+
+Given('I have hooks adding authorization to "POST /articles"', function () {
+  this.dredd.args.push(`--hookfiles=test/fixtures/blog/hooks${this.hooksExt}`);
+});
+
+When('I run Dredd', { timeout: 10 * 1000 }, function (callback) {
```

Comment:
Ideally, this wouldn't need to be an end-to-end test. We could call `new Dredd()`, then run it and capture what happens through a custom `EventEmmiter`, but I wasn't really in a mood to get it working now as Dredd's configuration through JavaScript is ugly. This is an initial proof-of-concept of this approach and we'll need e2e tests later as well anyway, so this code is going to be useful.

Reply:
Not possible now, because of https://github.com/apiaryio/dredd/issues/1424